src/Configuration/src/Kubernetes.ServiceBinding/PostProcessors.cs

```diff
@@ -590,21 +590,15 @@ public void PostProcessConfiguration(PostProcessorConfigurationProvider provider
         configurationData.Filter(ServiceBindingConfigurationProvider.InputKeyPrefix, ServiceBindingConfigurationProvider.TypeKey, BindingType).ForEach(
             bindingNameKey =>
             {
-                // Spring -> spring.datasource....
-                // Steeltoe -> steeltoe:service-bindings:sqlserver:binding-name:....
                 var mapper = new ServiceBindingMapper(configurationData, bindingNameKey, ServiceBindingConfigurationProvider.OutputKeyPrefix, BindingType,
                     ConfigurationPath.GetSectionKey(bindingNameKey));
 
-                mapper.MapFromTo("username", "username");
-                mapper.MapFromTo("password", "password");
-                mapper.MapFromTo("host", "host");
-                mapper.MapFromTo("port", "port");
-                mapper.MapFromTo("database", "database");
-
-                // Spring indicates this takes precedence over above
```

Comment:
Yes, it is. We've discussed whether these post-processors should be general-purpose or specific to their consuming connectors and decided to go with the latter for simplicity. The produced configuration keys are tailored to their corresponding connector, so that connectors can leverage driver-supplied connection string parsers, such as  [SqlConnectionStringBuilder](https://learn.microsoft.com/en-us/dotnet/api/microsoft.data.sqlclient.sqlconnectionstringbuilder?view=sqlclient-dotnet-core-5.1), [NpgsqlConnectionStringBuilder](https://github.com/npgsql/npgsql/blob/main/src/Npgsql/NpgsqlConnectionStringBuilder.cs), [MySqlConnectionStringBuilder](https://github.com/mysql-net/MySqlConnector/blob/master/src/MySqlConnector/MySqlConnectionStringBuilder.cs), etc. This means, for example, that the hostname for SQL Server is exposed as "Data Source", and it includes the port number. In contrast to PostgreSQL, where they map to "host" and "port". If we'd have generic keys, we'd need another translation layer to map the generic keys into connector-specific keys. Also, the data types may differ, where one driver expects a boolean and another expects an enumeration.

We've also discussed whether we should preserve all existing mapping logic (targeting products that we know no longer exist, and therefore we cannot test for correctness), or reduce the scope to an initial small set of supported brokers with full test coverage, samples, docs, etc. We've decided to start small and simple, then later expand on that as the need arises. For SQL Server, the new connectors currently only support Azure SQL Server broker on CF, which does not need `jdbc-url`, which is why I've removed it. If a future broker requires it, we'll add it back in and adapt its connector to consume it accordingly.

On a side note, we've had several issues with `jdbc-url`, in terms of invalid URIs due to non-standard comma occurrences and incorrect character escaping in specific brokers. In the Java ecosystem, many drivers take a `jdbc-url` as input, so it's convenient to use that directly. However, none of the drivers in .NET can handle them. In general, the separate values are more reliable, so I'm trying to stay away from these URLs unless required.

Dave did an initial rough port of the Java post-processors, adding various comments for additional concerns. As I'm writing/testing the connectors that consume these keys, I'm doing another pass over these post-processors (and ensuring they match the CF equivalents I'm adding), while adapting them to what the connectors expect.

Reply:
Ok this makes sense.